### PR TITLE
Add Evolution whitelist support

### PR DESCRIPTION
**Pull request by SonnyX**

### CODE REVIEW

File: src/main/java/dekk/pw/pokemate/tasks/EvolvePokemon.java [hunk 1]
```diff
@@ -34,14 +35,16 @@
...
             for (Pokemon pokemon : context.getApi().getInventories().getPokebank().getPokemons()) {
-                int number = pokemon.getPokemonId().getNumber();
-                if (CANDY_AMOUNTS.containsKey(number)) {
-                    int required = CANDY_AMOUNTS.get(number);
-                    if (required < 1) continue;
-                    if (pokemon.getCandy() >= required) {
-                        EvolutionResult result = pokemon.evolve();
-                        if (result.isSuccessful()) {
-                            System.out.println(pokemon.getPokemonId() + " has evolved into " + result.getEvolvedPokemon().getPokemonId() + " costing " + required + " candies");
+                if (Config.isWhitelistEnabled() ? Config.getWhitelistedPokemons().contains(pokemon.getId()) : true) {
```
Comment (kyleshrader) [inline src/main/java/dekk/pw/pokemate/tasks/EvolvePokemon.java:38]: What is the result of pokemon.getId()? Is that the unique id or is it the pokemon's number?

Comment (SwipeX) [inline src/main/java/dekk/pw/pokemate/tasks/EvolvePokemon.java:38]: It is a wrapper, number is the number

File: src/main/java/dekk/pw/pokemate/Config.java [hunk 2]
```diff
@@ -90,6 +96,19 @@ public static Properties getProperties() {
     public static boolean isAutoEvolving() {
         return autoEvolving;
     }
+    
+    public static boolean isWhitelistEnabled() {
+        return whitelistEnabled;
+    }
+    
+    public static List<Long> getWhitelistedPokemons() {
```
Comment (SwipeX) [inline src/main/java/dekk/pw/pokemate/Config.java:104]: Not sure I agree with longs, considering there are 151 pokemon.

**Comment by SwipeX** [comment]
Im liking the idea, but I dont think this is the way to go about it. Any thoughts? Also see my comment on longs. Integers may be fine, but a short would work as well.

**Comment by kyleshrader** [comment]
Agreed; I think this is something that should be added after the config is in the UI or a config editor is in place... It is a pretty heavy item for config - at least from a user perspective

**Comment by SonnyX** [comment]
The idea of this whitelist is also not to add all 151 pokemons, if that were the case you would just disable the whitelist, I assume most people would just add three to twenty, pretty much the regular pokemons.
I wrote this to prevent the script from evolving pokemons sush as magicarp, dratini, and other pokemons that are rare and/or require a lot of candy.

**Comment by SwipeX** [comment]
@SonnyX correct.  3-20 sounds reasonable.

**Comment by SwipeX** [comment]
@SonnyX  don't forget Pokemon#pokemonid#pokemonNumber is what will give the integer.  Submit a new PR with the changes and ill gladly accept.

**Pull request closed without merging**